### DISCUSSION TITLE
Pytorch lightning DDP vs DDP_SPAWN

### DISCUSSION THREAD

**Discussion by senzeyu** [optuna/optuna#4837] [Q&A]
### Don't use GitHub Issues to ask support questions.

I see the optuna's [example on pytorch_lightning's DDP_SPAWN](https://github.com/optuna/optuna-examples/blob/main/pytorch/pytorch_lightning_ddp.py#L136C19-L136C28). Does Optuna support DDP? 

When I use DDP instead of DDP_SPAWN, it seems like each GPU's process tries to create a trial. Is this expected behavior and how may I fix that?

**Comment by Akshay1-6180** [comment]
I tried both the difference between ddp and ddp_spawn is that , in the case of 2 gpus
1) ddp creates 2 seperate process to train the model for 2 seperate trials
2) ddp_spawn uses both the gpus for a single trial but is very slow compared to ddp , not sure if there is support for ddp , maybe @nzw0301 can answer

**Reply by nzw0301** [reply]
Hello, thank you for letting me know this. I'm afraid that I do not have multiple GPUs, so I cannot investigate this, sorry.
